packages/service-library/src/servicelib/fastapi/docker.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+import logging
+from collections.abc import AsyncIterator
+from contextlib import AsyncExitStack
+from typing import Final
+
+import aiodocker
+import aiohttp
+import tenacity
+from aiohttp import ClientSession
+from fastapi import FastAPI
+from fastapi_lifespan_manager import State
+from pydantic import NonNegativeInt
+from servicelib.fastapi.lifespan_utils import LifespanGenerator
+from settings_library.docker_api_proxy import DockerApiProxysettings
+
+_logger = logging.getLogger(__name__)
+
+_DEFAULT_DOCKER_API_PROXY_HEALTH_TIMEOUT: Final[NonNegativeInt] = 5
+
+
+def get_lifespan_remote_docker_client(
```

Comment:
why is this one called differently than the other lifespan_* methods? why get_?

Reply:
Because this returns the LifespanGenerator, it's not the LifespanGenerator